src/cli/format-results-cache.js

```diff
@@ -0,0 +1,91 @@
+"use strict";
+
+// Inspired by LintResultsCache from ESLint
+// https://github.com/eslint/eslint/blob/c2d0a830754b6099a3325e6d3348c3ba983a677a/lib/cli-engine/lint-result-cache.js
+
+const fileEntryCache = require("file-entry-cache");
+const stringify = require("fast-json-stable-stringify");
+// eslint-disable-next-line no-restricted-modules
+const { version: prettierVersion } = require("../index.js");
+const { createHash } = require("./utils.js");
+
+const optionsHashCache = new WeakMap();
+const nodeVersion = process && process.version;
+
+/**
+ * @param {*} options
+ * @returns {string}
+ */
+function getHashOfOptions(options) {
+  if (optionsHashCache.has(options)) {
+    return optionsHashCache.get(options);
+  }
+  const hash = createHash(
+    `${prettierVersion}_${nodeVersion}_${stringify(options)}`
+  );
+  optionsHashCache.set(options, hash);
+  return hash;
+}
+
+/**
+ * @typedef {{ hashOfOptions?: string }} OurMeta
+ * @typedef {import("file-entry-cache").FileDescriptor} FileDescriptor
+ *
+ * @param {import("file-entry-cache").FileDescriptor} fileDescriptor
+ * @returns {FileDescriptor["meta"] & OurMeta}
+ */
+function getMetadataFromFileDescriptor(fileDescriptor) {
+  return fileDescriptor.meta;
+}
```

Comment:
Why don't we use `fileDescriptor.meta` directly? This just making code less readable.

Reply:
Just for type checking...